### PR TITLE
✨ Use RollingUpdate strategy for cron worker job

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Adds `RollingUpdate` strategy to cron worker job to allow for auto updates.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: cron/k8s/worker.yaml [hunk 1]
```diff
@@ -47,3 +47,8 @@ spec:
           limits:
             cpu: 12
             memory: 12Gi
+  strategy:
+    type: "RollingUpdate"
+    rollingUpdate:
+      maxUnavailable: 1
```
Comment (naveensrinivasan) [inline cron/k8s/worker.yaml:53]: If there is only one replica this will not work. We need at least 2 to rolling update to work.

Comment (azeemshaikh38) [reply cron/k8s/worker.yaml:53]: Doesn't this imply a `n+1` RollingUpdate strategy here? Basically, what I want to have here is a strategy that brings up a new instance (`maxSurge` = 1) and kills an older one (`maxUnavailable` = 1).

Let me know if this isn't doing what I expect it to do and what change I need to make here for it work that way.

Comment (naveensrinivasan) [reply cron/k8s/worker.yaml:53]: Have you tried this? I am not aware of the max surge with this option. As long as you have tried it and it works I am fine.

Comment (azeemshaikh38) [reply cron/k8s/worker.yaml:53]: I tried it through the UI and it worked. But then again, we only have 1 worker instance running. 

Maybe we keep this as is for now? My main aim is unblock auto deployment of new binaries when they become available. If the behavior breaks with >1 worker instances, we can revisit then. Wdyt?

Comment (naveensrinivasan) [reply cron/k8s/worker.yaml:53]: Sounds good.

**Review by naveensrinivasan — APPROVED** [review]

**Pull request merged**